Tensile/Components/LocalRead.py

```diff
@@ -227,55 +224,62 @@ def __call__(self, writer, bufferIdx, iui, epsi, tP):
+                          divVal = tP["bpe"] * lrvw // writer.bpr
                           rIdxMod = rIdx % divVal
                           rIdxDiv = rIdx // divVal
-                          offset_val = (rIdxDiv * UnrollStride + offset_val + localReadOffset) * tP["bpe"]  + rIdxMod * writer.bpr
+                          offset_val = (rIdxDiv * UnrollStride + offset_val + localReadOffset) * tP["bpe"] * blockWidth + rIdxMod * writer.bpr
                         else:
                           # normal case
                           offset_val = (rIdx * UnrollStride + offset_val + localReadOffset) * tP["bpe"]
-                        if kernel["ThreadSeparateGlobalRead%c"%tc] and localReadOffsetDiv > 0:
+                        if localReadOffsetDiv > 0:
                           # TSGR special conversion
                           # Multiply BlockSize for each lrdOffsetMod
-                          MblockSizePerLoad = (kernel["WavefrontSize"] * kernel["GlobalLoadVectorWidth%c"%tc]) // kernel["_DepthULds"]
+                          MblockSizePerLoad = (kernel["WavefrontSize"] * kernel["GlobalLoadVectorWidth%c"%tc]) // ldsWidth
                           offset_val += ((MblockSizePerLoad * lrdOffsetMod * tP["bpe"]) * localReadOffsetDiv)
                         if (kernel["LdsBlockSizePerPad%s"%tc] != 0) and (kernel["LdsPad%s"%tc] != 0):
                             offset_val = offset_val + (offset_val // kernel["LdsBlockSizePerPad%s"%tc]) * kernel["LdsPad%s"%tc] * tP["bpe"]
                         offset_val = offset_val + tP["localReadSwapByteOffset"]
-                        if (kernel["DirectToLds%s" % tc] and  \
-                            kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"] > 4):
-
+                        if kernel["DirectToLds%s" % tc]:
                           # another address conversion for DirectToLds + NumLoadsCoalesced > 1
                           dummy, offset_val = writer.lraOffsetConversionForDTLandNLC(kernel, tP, offset_val)
 
-                          # offset conversion for DirectToLds
-                          # TLU=0 case, modify bit3-6 of offset_val as follows
-                          # (bit2<<3) | (bit3 <<1) | (bit4>>2) | (bit5>>2)
-                          bit2 = offset_val & 4
-                          bit3 = offset_val & 8
-                          bit4 = offset_val & 16
-                          bit5 = offset_val & 32
-                          if (kernel["GlobalLoadVectorWidth%s"%tc] * tP["bpe"] == 8):
-                            # dword_x2 case
-                            # (bit2<<3) | (bit3 >>1) | (bit4>>1) | (bit5>>1)
-                            newVal = (bit2<<3) | (bit3 >>1) | (bit4>>1) | (bit5>>1)
-                          else:  #if (kernel["GlobalLoadVectorWidth%s"%tc] * tP["bpe"] == 16):  # most preferred case
-                            # dword_x4 case
+                          if kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"] > 4:
```

Comment:
This looks like the case that is no longer supported, right? We should probably mark as unavailable in current ISA.

Reply:
I kept this logic just in case the assembler supports this again later.
If necessary, we can clean up the code later.